### PR TITLE
fix callbacks on release version

### PR DESCRIPTION
**Pull request by cirospaciari**

### CODE REVIEW

File: src/bun.js/api/bun/socket.zig [hunk 1]
```diff
@@ -2114,11 +2113,17 @@ fn NewSocket(comptime ssl: bool) type {
...
                 .onHandshake = this.handlers.onHandshake,
                 .binary_type = this.handlers.binary_type,
+                .is_server = this.handlers.is_server,
             };
-
-            raw_handlers_ptr.* = cloned_handlers;
-            raw_handlers_ptr.is_server = this.handlers.is_server;
-            raw_handlers_ptr.protect();
+            this.handlers.onOpen = .zero;
+            this.handlers.onClose = .zero;
```
Comment (Jarred-Sumner) [inline src/bun.js/api/bun/socket.zig:2119]: is the idea here that we detach the previous handlers?

File: src/bun.js/api/bun/socket.zig [hunk 2]
```diff
@@ -1972,7 +1972,11 @@ fn NewSocket(comptime ssl: bool) type {
             _: *JSC.CallFrame,
         ) callconv(.C) JSValue {
             JSC.markBinding(@src());
-            this.socket.open(!this.handlers.is_server);
+            if (comptime ssl) {
+                if (!this.detached) {
+                    this.socket.open(!this.handlers.is_server);
```
Comment (Jarred-Sumner) [inline src/bun.js/api/bun/socket.zig:1977]: I don't 100% understand the purpose of this function to begin with

File: src/bun.js/api/bun/socket.zig (hunk 1 above)
Comment (cirospaciari) [reply src/bun.js/api/bun/socket.zig:2119]: yes, this function basically detach the current socket and create 2 new ones, one raw and one TLS

File: src/bun.js/api/bun/socket.zig (hunk 2 above)
Comment (cirospaciari) [reply src/bun.js/api/bun/socket.zig:1977]: this basically starts TLS handshake, in a control manner after we deal with JS stuff internal socket attach

**Pull request merged**